src/Moka/Proxy/Proxy.php

```diff
@@ -35,9 +35,9 @@ public function serve(): MockObject
 
     /**
      * @param array $methodsWithValues
-     * @return Proxy
+     * @return ProxyInterface
      */
-    public function stub(array $methodsWithValues): self
+    public function stub(array $methodsWithValues): ProxyInterface
```

Comment:
You simply do not have that choice, because changing the return will break the interface; it's a PHP limitation.